src/backend/distributed/utils/reference_table_utils.c

```diff
@@ -207,6 +207,13 @@ EnsureReferenceTablesExistOnAllNodesExtended(char transferMode)
 				CopyShardPlacementToWorkerNodeQuery(sourceShardPlacement,
 													newWorkerNode,
 													transferMode);
+
+			/*
+			 * The placement copy command uses distributed execution to copy
+			 * the shard, this is all fine so we temporarily allow it.
+			 */
+			ExecuteCriticalRemoteCommand(connection,
+										 "SET LOCAL citus.allow_nested_distributed_execution = true");
```

Comment:
in the rebalancer we do this by setting the application_name:
```
    StringInfo setApplicationName = makeStringInfo();
    appendStringInfo(setApplicationName, "SET application_name TO %s",
                     CITUS_REBALANCER_NAME);
```
wondering whether we should do something similar here (maybe generalized the application_name)

Reply:
I changed this to use the same application_name trick, but I don't think it's worth spending time on to make this more generic. Since the rebalancer won't run commands like this anymore in the near future, because of Nils his background daemon changes.